[1.0.0rc2] test_yanny fails on big endian machines

On a Debian system with s90x architecture (IBM zSystems, big endian, 64 bit), we get the following test failure:
```
_____________________________ TestYanny.test_yanny _____________________________

self = <pydl.pydlutils.tests.test_yanny.TestYanny object at 0x3ff377b37c0>

        def test_yanny(self):
            """Used to test the yanny class.
            """
            #[…]
            for t in par.tables():
>               assert (par.dtype(t) ==
                        self.json2dtype(struct_data[t]['dtype']))
E                       AssertionError: assert dtype([('mag', '>f4', (5,)), ('b', 'S33', (5,)), ('foo', 'S25'), ('c', '>f8'), ('flags', '>i4', (2,)), ('new_flag', 'S5')]) == dtype([('mag', '<f4', (5,)), ('b', 'S33', (5,)), ('foo', 'S25'), ('c', '<f8'), ('flags', '<i4', (2,)), ('new_flag', 'S5')])
E                        +  where dtype([('mag', '>f4', (5,)), ('b', 'S33', (5,)), ('foo', 'S25'), ('c', '>f8'), ('flags', '>i4', (2,)), ('new_flag', 'S5')]) = <bound method yanny.dtype of #%yanny\n# test.par\n#\n# FTCL/Yanny file for testing the perl & python yanny readers.\n#\n# T...ESS 2008-06-22T00:27:33\nSTATUS_UPDATE FAILURE 2008-06-22T01:27:33\n#\nold {1.2 2.3 3.4} hello\nold {4.3 3.2 2.1} { { } }\n>('MYSTRUCT')
E                        +    where <bound method yanny.dtype of #%yanny\n# test.par\n#\n# FTCL/Yanny file for testing the perl & python yanny readers.\n#\n# T...ESS 2008-06-22T00:27:33\nSTATUS_UPDATE FAILURE 2008-06-22T01:27:33\n#\nold {1.2 2.3 3.4} hello\nold {4.3 3.2 2.1} { { } }\n> = #%yanny\n# test.par\n#\n# FTCL/Yanny file for testing the perl & python yanny readers.\n#\n# This file is much more complic...CESS 2008-06-22T00:27:33\nSTATUS_UPDATE FAILURE 2008-06-22T01:27:33\n#\nold {1.2 2.3 3.4} hello\nold {4.3 3.2 2.1} { { } }\n.dtype
E                        +  and   dtype([('mag', '<f4', (5,)), ('b', 'S33', (5,)), ('foo', 'S25'), ('c', '<f8'), ('flags', '<i4', (2,)), ('new_flag', 'S5')]) = <bound method YannyTestCase.json2dtype of <pydl.pydlutils.tests.test_yanny.TestYanny object at 0x3ff377b37c0>>([['mag', '<f4', [5]], ['b', 'S33', [5]], ['foo', 'S25'], ['c', '<f8'], ['flags', '<i4', [2]], ['new_flag', 'S5']])
E                        +    where <bound method YannyTestCase.json2dtype of <pydl.pydlutils.tests.test_yanny.TestYanny object at 0x3ff377b37c0>> = <pydl.pydlutils.tests.test_yanny.TestYanny object at 0x3ff377b37c0>.json2dtype

/usr/lib/python3/dist-packages/pydl/pydlutils/tests/test_yanny.py:300: AssertionError
```
Full test log [here](https://ci.debian.net/data/autopkgtest/testing/s390x/p/pydl/30548623/log.gz).
I think this is a wrong endianness mark somewhere.

--- a/.github/workflows/ci_cron_weekly.yml
+++ b/.github/workflows/ci_cron_weekly.yml
@@ -21,7 +21,7 @@ jobs:
     - name: Install base dependencies
       run: |
         python -m pip install --upgrade pip
-        python -m pip install tox\<4
+        python -m pip install "tox<4"
     - name: Check links in docs using tox
       run: |
         tox -e linkcheck

--- a/.github/workflows/ci_cron_weekly.yml
+++ b/.github/workflows/ci_cron_weekly.yml
@@ -11,7 +11,7 @@ jobs:
   # Testing links in documents is a good example of something to run on a schedule
   # to catch links that stop working for some reason.
   doc_test:
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
     - name: Set up Python to build docs with sphinx
@@ -21,7 +21,7 @@ jobs:
     - name: Install base dependencies
       run: |
         python -m pip install --upgrade pip
-        python -m pip install "tox<4"
+        python -m pip install tox
     - name: Check links in docs using tox
       run: |
         tox -e linkcheck

--- a/.github/workflows/ci_cron_weekly.yml
+++ b/.github/workflows/ci_cron_weekly.yml
@@ -21,7 +21,7 @@ jobs:
     - name: Install base dependencies
       run: |
         python -m pip install --upgrade pip
-        python -m pip install tox
+        python -m pip install tox\<4
     - name: Check links in docs using tox
       run: |
         tox -e linkcheck

--- a/.github/workflows/ci_tests.yml
+++ b/.github/workflows/ci_tests.yml
@@ -74,7 +74,7 @@ jobs:
     - name: Install base dependencies
       run: |
         python -m pip install --upgrade pip
-        python -m pip install tox coveralls
+        python -m pip install tox\<4 coveralls
     - name: Install graphviz dependency
       if: "endsWith(matrix.toxenv, 'build_docs')"
       run: sudo apt-get -y install graphviz

--- a/.github/workflows/ci_tests.yml
+++ b/.github/workflows/ci_tests.yml
@@ -37,7 +37,7 @@ jobs:
             python: 3.8
             toxenv: py38-test-alldeps-cov
 
-          - name: OS X - Python 3.8 with all optional dependencies
+          - name: macOS - Python 3.8 with all optional dependencies
             os: macos-latest
             python: 3.8
             toxenv: py38-test-alldeps
@@ -74,7 +74,7 @@ jobs:
     - name: Install base dependencies
       run: |
         python -m pip install --upgrade pip
-        python -m pip install "tox<4" coveralls
+        python -m pip install tox coveralls
     - name: Install graphviz dependency
       if: "endsWith(matrix.toxenv, 'build_docs')"
       run: sudo apt-get -y install graphviz

--- a/.github/workflows/ci_tests.yml
+++ b/.github/workflows/ci_tests.yml
@@ -74,7 +74,7 @@ jobs:
     - name: Install base dependencies
       run: |
         python -m pip install --upgrade pip
-        python -m pip install tox\<4 coveralls
+        python -m pip install "tox<4" coveralls
     - name: Install graphviz dependency
       if: "endsWith(matrix.toxenv, 'build_docs')"
       run: sudo apt-get -y install graphviz

--- a/.gitignore
+++ b/.gitignore
@@ -36,6 +36,7 @@ docs/_build
 .flooignore
 
 # Visual Studio Code project files
+.env
 .vscode
 
 # Packages/installer info

--- a/.readthedocs.yml
+++ b/.readthedocs.yml
@@ -1,16 +1,27 @@
-# See https://docs.readthedocs.io/en/latest/yaml-config.html
+# .readthedocs.yaml
+# Read the Docs configuration file
+# See https://docs.readthedocs.io/en/stable/config-file/v2.html for details
+
+# Required
 version: 2
 
 build:
-    image: latest
+  os: ubuntu-22.04
+  tools:
+    python: "3.11"
 
+# Build documentation in the docs/ directory with Sphinx
+sphinx:
+  configuration: docs/conf.py
+
+# Optionally declare the Python requirements required to build your docs
 python:
-    version: 3.7
-    install:
-        - method: pip
-          path: .
-          extra_requirements:
-                - docs
-                - all
+  install:
+    - method: pip
+      path: .
+      extra_requirements:
+        - docs
+        - all
 
+# If using Sphinx, optionally build your docs in additional formats such as PDF
 formats: []

--- a/pydl/pydlspec2d/spec2d.py
+++ b/pydl/pydlspec2d/spec2d.py
@@ -436,7 +436,7 @@ def filter_thru(flux, waveimg=None, wset=None, mask=None,
     pixnorm, logdiff = traceset2xy(diffset)
     logdiff = np.absolute(logdiff)
     if mask is not None:
-        flux_interp = djs_maskinterp(flux, mask, iaxis=0)
+        flux_interp = djs_maskinterp(flux, mask, axis=0)
     res = np.zeros((nTrace, len(ffiles)), dtype=flux.dtype)
     for i, f in enumerate(ffiles):
         filter_data = ascii.read(f, comment='#.*', names=('lam', 'respt',

--- a/pydl/pydlutils/tests/test_mangle.py
+++ b/pydl/pydlutils/tests/test_mangle.py
@@ -11,15 +11,12 @@ class TestMangle(object):
     """Test the functions in pydl.pydlutils.mangle.
     """
 
-    def setup(self):
+    def setup_method(self):
         self.poly_fits = get_pkg_data_filename('t/polygon.fits')
         self.no_id_fits = get_pkg_data_filename('t/polygon_no_id.fits')
         self.one_cap_fits = get_pkg_data_filename('t/polygon_one_cap.fits')
         self.poly_ply = get_pkg_data_filename('t/polygon.ply')
         self.bad_ply = get_pkg_data_filename('t/median_data.txt')
-
-    def teardown(self):
-        pass
 
     def test_ManglePolygon(self):
         #

--- a/pydl/pydlutils/tests/test_sdss.py
+++ b/pydl/pydlutils/tests/test_sdss.py
@@ -13,7 +13,7 @@ class TestSDSS(object):
     """Test the functions in pydl.pydlutils.sdss.
     """
 
-    def setup(self):
+    def setup_method(self):
         pydl.pydlutils.sdss.maskbits = set_maskbits(
             maskbits_file=get_pkg_data_filename('t/testMaskbits.par'))
         self.opbadfields = np.array([
@@ -27,9 +27,6 @@ class TestSDSS(object):
             dtype=[('run', '<i4'), ('problem', 'S8'), ('firstfield', '<i4'), ('lastfield', '<i4'), ('comments', 'S47')])
         pydl.pydlutils.sdss.opbadfields = self.opbadfields
         return
-
-    def teardown(self):
-        pass
 
     def test_default_skyversion(self):
         assert default_skyversion() == 2

--- a/pydl/pydlutils/tests/test_trace.py
+++ b/pydl/pydlutils/tests/test_trace.py
@@ -13,7 +13,7 @@ class TestTrace(object):
     """Test the functions in pydl.pydlutils.trace.
     """
 
-    def setup(self):
+    def setup_method(self):
         # extracted from spFrame-b1-00057618.fits
 
         self.sdss = fits.open(get_pkg_data_filename('t/sdss_traceset.fits'))
@@ -21,7 +21,7 @@ class TestTrace(object):
         self.boss = fits.open(get_pkg_data_filename('t/boss_traceset.fits'))
         return
 
-    def teardown(self):
+    def teardown_method(self):
         self.sdss.close()
         self.boss.close()
         return

--- a/pydl/pydlutils/tests/test_yanny.py
+++ b/pydl/pydlutils/tests/test_yanny.py
@@ -2,6 +2,8 @@
 # -*- coding: utf-8 -*-
 import warnings
 import json
+import stat
+import sys
 from os import chmod, remove
 from os.path import exists, join
 from shutil import copy, rmtree
@@ -82,7 +84,11 @@ class YannyTestCase(object):
                 stuff.append(tuple([str(k[0]), str(k[1]), tuple(k[2])]))
             else:
                 stuff.append(tuple([str(k[0]), str(k[1])]))
-        return np.dtype(stuff)
+        d = np.dtype(stuff)
+        if sys.byteorder == 'little':
+            return d
+        else:
+            return d.newbyteorder()
 
 
 class TestYanny(YannyTestCase):

--- a/pydl/pydlutils/tests/test_yanny.py
+++ b/pydl/pydlutils/tests/test_yanny.py
@@ -31,7 +31,7 @@ class YannyTestCase(object):
     """
     save_temp = False
 
-    def setup(self):
+    def setup_method(self):
         self.temp_dir = mkdtemp(prefix='yanny-test-')
         # Ignore deprecation warnings--this only affects Python 2.5 and 2.6,
         # since deprecation warnings are ignored by defualt on 2.7
@@ -41,7 +41,7 @@ class YannyTestCase(object):
         with open(get_pkg_data_filename("t/yanny_data.json")) as js:
             self.test_data = json.load(js)
 
-    def teardown(self):
+    def teardown_method(self):
         warnings.resetwarnings()
         if not self.save_temp:
             if hasattr(self, 'temp_dir') and exists(self.temp_dir):

--- a/setup.cfg
+++ b/setup.cfg
@@ -42,8 +42,6 @@ all =
 test =
     pytest-astropy
     pytest-mock
-astropy20 =
-    pytest-mock<2
 docs =
     sphinx-astropy
 

--- a/tox.ini
+++ b/tox.ini
@@ -3,7 +3,6 @@ envlist =
     py{36,37,38}-test{,-alldeps,-devdeps}{,-cov}
     py{36,37,38}-test-numpy{116,117,118}
     py{36,37,38}-test-astropy{30,40,lts}
-    astropy20
     build_docs
     linkcheck
     codestyle
@@ -11,20 +10,16 @@ requires =
     setuptools >= 30.3.0
     pip >= 19.3.1
     setuptools_scm
-    tox < 4
-    tox-pypi-filter >= 0.12
 isolated_build = true
+indexserver =
+    NIGHTLY = https://pypi.anaconda.org/scipy-wheels-nightly/simple
 
 [testenv]
-
-# The following option combined with the use of the tox-pypi-filter above allows
-# project-wide pinning of dependencies, e.g. if new versions of pytest do not
-# work correctly with pytest-astropy plugins. Most of the time the pinnings file
-# should be empty.
-pypi_filter = https://raw.githubusercontent.com/astropy/ci-helpers/master/pip_pinnings.txt
+# Suppress display of matplotlib plots generated during docs build
+setenv = MPLBACKEND=agg
 
 # Pass through the following environment variables which may be needed for the CI
-passenv = HOME WINDIR LC_ALL LC_CTYPE CC CI GITHUB_* COVERALLS_*
+passenv = HOME,WINDIR,LC_ALL,LC_CTYPE,CC,CI,GITHUB_*,COVERALLS_*
 
 # Run the tests in a temporary directory to make sure that we don't import
 # this package from the source tree
@@ -53,6 +48,7 @@ description =
 
 # The following provides some specific pinnings for key packages
 deps =
+
     cov: coveralls
 
     numpy116: numpy==1.16.*
@@ -63,7 +59,7 @@ deps =
     astropy40: astropy==4.0.*
     astropylts: astropy==4.0.*
 
-    devdeps: git+https://github.com/numpy/numpy.git#egg=numpy
+    devdeps: :NIGHTLY:numpy
     devdeps: git+https://github.com/astropy/astropy.git#egg=astropy
 
 # The following indicates which extras_require from setup.cfg will be installed
@@ -76,18 +72,6 @@ commands =
     !cov: pytest --pyargs pydl {toxinidir}/docs {posargs}
     cov: pytest --pyargs pydl {toxinidir}/docs --cov pydl --cov-config={toxinidir}/setup.cfg {posargs}
     cov: coveralls
-
-[testenv:astropy20]
-description = run tests with astropy2.0.*
-extras = astropy20
-deps =
-    astropy==2.0.*
-    numpy==1.14.*
-    pytest<3.7
-    pytest-mock<2
-commands =
-    pip freeze
-    pytest -p astropy.tests.pytest_repeat -p astropy.extern.plugins.pytest_doctestplus.plugin -p astropy.extern.plugins.pytest_openfiles.plugin -p astropy.extern.plugins.pytest_remotedata.plugin --traceconfig --pyargs pydl {toxinidir}/docs {posargs}
 
 [testenv:build_docs]
 changedir = docs

--- a/tox.ini
+++ b/tox.ini
@@ -11,6 +11,7 @@ requires =
     setuptools >= 30.3.0
     pip >= 19.3.1
     setuptools_scm
+    tox < 4
     tox-pypi-filter >= 0.12
 isolated_build = true
 

